android/app/src/main/java/poke/rogue/helper/presentation/dex/detail/BattlePopUpUiModel.kt

```diff
@@ -0,0 +1,28 @@
+package poke.rogue.helper.presentation.dex.detail
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import poke.rogue.helper.R
+
+sealed class BattlePopUpUiModel(
+    @StringRes val description: Int,
+    @DrawableRes val icon: Int,
+) {
+    companion object {
+        val items =
+            listOf(
+                EnemyPokemon(
+                    R.string.pokemon_detail_to_battle_with_enemy_pop_up_descriptions,
+                    R.drawable.ic_pokemon_battle_enemy,
+                ),
+                MyPokemon(
+                    R.string.pokemon_detail_to_battle_with_mine_pop_up_descriptions,
+                    R.drawable.ic_pokemon_battle_mine,
+                ),
+            )
+    }
+}
+
+class MyPokemon(description: Int, icon: Int) : BattlePopUpUiModel(description, icon)
```

Comment:
```suggestion
class MyPokemonUiModel(description: Int, icon: Int) : BattlePopUpUiModel(description, icon)
```

UiModel 에는 `UiModel` postfix를 붙여줘야합니다!  

[네이밍 컨벤션](
https://github.com/woowacourse-teams/2024-pokerogue-helper/wiki/%F0%9F%A4%96-AN-%EA%B0%9C%EB%B0%9C-%EC%BB%A8%EB%B2%A4%EC%85%98#-naming)